test/test_datatype.jl

```diff
@@ -0,0 +1,48 @@
+using Base.Test
+using MPI
+
+MPI.Init()
+
+immutable Boundary
+  c::UInt16  # force some padding to be inserted
+  a::Int
+  b::UInt8
+end
```

Comment:
Does this also work with tuples, i.e. if you were to add `d::Tuple{Int,Float64}` here?